### ISSUE TITLE
wish to print a game

### ISSUE THREAD

**Issue by gijsbers** [lukasmonk/lucaschessR2#117]
One can analyse a game and save it, but there is no print option. After analysis I want to print it and go to a real board to study it. For that the PGN needs to be properly formatted. A bit like this:
![20](https://github.com/lukasmonk/lucaschessR2/assets/4169687/08d35438-6244-4fe0-a10d-432427a7cfc2)
But of course in practice the analysis will be much more extensive. This facility could also be helpful for imported PGNs. Saving the formatted print to a PDF would also be nice.

**Comment by gijsbers** [comment]
Here is a better example of what I hope to see in printed game analysis: a tree of thinking.
![a-tree-of-thinking](https://github.com/lukasmonk/lucaschessR2/assets/4169687/d2081851-b06e-4163-842b-49185945687a)

**Comment by lukasmonk** [comment]
Thanks for the ideas.
I guess there must be a lot of programmes that make a manageable printout of a game. 
Maybe next year I'll get around to working on this.

**Comment by gijsbers** [comment]
Another wish. On Linux it is customary to pass command-line arguments to the program. Like pass the file location of a PGN file that can then be opened immediately for analysis. Like in: `Lucas file.pgn`. That way one doesn't have to navigate through the menu system with the mouse, which is much slower than the keyboard.

**Comment by lukasmonk** [comment]
This utility is already included.

**Comment by gijsbers** [comment]
OK, now I see it. Sofar I only got: `Failed to execute script 'Launcher' due to unhandled exception!` whenever I tried this, but if I give the absolute path it works. Great! Lucas changes its working directory when it starts, which causes relative paths not to be found.

**Comment by lukasmonk** [comment]
In a virtual machine with Debian, I have tested and it has let me assign Lucas Chess, as a program to be launched when double clicking on a pgn.

**Comment by gijsbers** [comment]
Apparently Debian then passes the full path to the PGN. In Linux if you are in terminal in a directory and see a file A, then you can say: `edit A` to open it in the editor. When you save the file to a different name B, then B will appear in that same directory. Here you don't need to use path separators like `/` or `\\`. But because Lucas changes directory to the Lucas installed bin subdirectory, the location information of a local PGN is lost if it is not a full path with all separators.

**Comment by lukasmonk** [comment]
You are right, support is only given when the full path is indicated.

**Comment by gijsbers** [comment]
And the `LucasR.sh` script on Linux doesn't pass on command-line arguments with `"$@"`.

The `bin/LucasR` program throws an unhandled exception on `-help` or `--help`,
which is a common way to ask a program for a synopsis of the options that it accepts.

**Comment by lukasmonk** [comment]
Lucas Chess is a graphical program designed in Windows and I try to make it work with the same possibilities in Linux, that can be installed and executed from a desktop, also that pgn files can be opened, through a file manager. In Linux the use of the terminal is more common than in Windows, and I don't intend to give more support for now.

**Comment by gijsbers** [comment]
When viewing an analysed game, the variations on the right show alternatives to the last move played. To understand this, I need to mentally undo the last move and then apply an alternative. This is cumbersome. I would much prefer the variations on the right to show me all the variation options I have as a move to play in the current position. Hence, the variations are one ply too late.

**Comment by lukasmonk** [comment]
How about a kibitzer of the candidate type?

**Comment by gijsbers** [comment]
That is so attention grabbing, because it changes all the time. If that is the only option, the mass analysis becomes obsolete. I seek to browse the mass analysis PGN in a way that shows me the analysis of the move options I have in the current position. Not having to mentally undo the last move before I can apply an analysis variation.

**Comment by lukasmonk** [comment]
It is likely that you have not understood what the real problem is.
But, are you talking about this?
![image](https://github.com/lukasmonk/lucaschessR2/assets/6254278/e4977e58-2650-4836-a09c-5e04d8451475)

**Comment by gijsbers** [comment]
Yes. The move d5 has already been executed, which makes looking at other variations more difficult. If the d-pawn would still be on the d4 square then examing variations would be more convenient.

**Comment by lukasmonk** [comment]
I don't know a good way to implement this.
With left click on the first move in each variation, the position on the board is changed.

**Comment by gijsbers** [comment]
The kibitzer window doesn't properly resize horizontally when I resize it to the full width of the screen. The column "Main line" remains the same and shows about 7 to 8 moves (w+b).

**Comment by lukasmonk** [comment]
This allows it to be enlarged manually.
![image](https://github.com/lukasmonk/lucaschessR2/assets/6254278/20bd1b20-8351-46f6-8ca4-755af02c648f)
